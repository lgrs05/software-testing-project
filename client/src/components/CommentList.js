import Comment from "./Comment";
import List from '@material-ui/core/List';
import PropTypes from 'prop-types';
import React from 'react';
import {withStyles} from '@material-ui/core/styles';


const styles = (theme) => ({
    root : {
        width           : '100%',
        maxWidth        : 360,
        backgroundColor : theme.palette.background.paper
    }
});

function CommentList(props) {
    const {classes: {root}, comments} = props;
    const getKey = ({name, message, time}) => `${name} ${message} ${time}`;
    return (
        <List className={root}>
            {comments.map((comment) =>
                <Comment key={getKey(comment)} {...comment} />)}
        </List>
    );
}

CommentList.propTypes = {
    classes  : PropTypes.object.isRequired,
    comments : PropTypes.array.isRequired
};

export default withStyles(styles)(CommentList);