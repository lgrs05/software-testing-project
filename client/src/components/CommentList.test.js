import CommentList from './CommentList';
import React from 'react';
import renderer from 'react-test-renderer';

let comments = [];

beforeEach(() => {

    comments = [
        {name    : 'Pedro',
            message : "This is a good message",
            time    : new Date(2019, 2, 21, 12, 0, 0, 0).toLocaleString()},
        {name    : 'Pedro Vazquez',
            message : "This is a great message",
            time    : new Date(2019, 2, 21, 12, 0, 0, 0).toLocaleString()},
        {name    : 'Pedro Santiago',
            message : "This is a bad message",
            time    : new Date(2019, 2, 21, 12, 0, 0, 0).toLocaleString()},
        {name    : 'Pedro Rivera',
            message : "This is an awesome message",
            time    : new Date(2019, 2, 21, 12, 0, 0, 0).toLocaleString()}
    ];
});

it('renders correctly', () => {
    const tree = renderer.create(<CommentList comments={comments}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});