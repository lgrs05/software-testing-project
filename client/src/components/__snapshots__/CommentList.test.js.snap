// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="MuiList-root-2 MuiList-padding-3 CommentList-root-1"
>
  <li
    className="MuiListItem-root-6 MuiListItem-default-9 MuiListItem-dense-10 MuiListItem-gutters-14 MuiListItem-alignItemsFlexStart-11"
    disabled={false}
  >
    <div
      className="MuiAvatar-root-21 MuiListItemAvatar-alignItemsFlexStart-19"
    >
      <img
        alt="Pedro"
        className="MuiAvatar-img-23"
        src="https://source.unsplash.com/2ShvY8Lf6l0/800x599"
      />
    </div>
    <div
      className="MuiListItemText-root-24"
    >
      <span
        className="MuiTypography-root-30 MuiTypography-subheading-37 MuiListItemText-primary-27"
      >
        Pedro
      </span>
      <p
        className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextSecondary-63 MuiListItemText-secondary-28"
      >
        <span
          className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextPrimary-62 MuiTypography-inline-65"
        >
          This is a good message
        </span>
        <br />
        2019-3-21 12:00:00
      </p>
    </div>
  </li>
  <li
    className="MuiListItem-root-6 MuiListItem-default-9 MuiListItem-dense-10 MuiListItem-gutters-14 MuiListItem-alignItemsFlexStart-11"
    disabled={false}
  >
    <div
      className="MuiAvatar-root-21 MuiListItemAvatar-alignItemsFlexStart-19"
    >
      <img
        alt="Pedro Vazquez"
        className="MuiAvatar-img-23"
        src="https://source.unsplash.com/2ShvY8Lf6l0/800x599"
      />
    </div>
    <div
      className="MuiListItemText-root-24"
    >
      <span
        className="MuiTypography-root-30 MuiTypography-subheading-37 MuiListItemText-primary-27"
      >
        Pedro Vazquez
      </span>
      <p
        className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextSecondary-63 MuiListItemText-secondary-28"
      >
        <span
          className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextPrimary-62 MuiTypography-inline-65"
        >
          This is a great message
        </span>
        <br />
        2019-3-21 12:00:00
      </p>
    </div>
  </li>
  <li
    className="MuiListItem-root-6 MuiListItem-default-9 MuiListItem-dense-10 MuiListItem-gutters-14 MuiListItem-alignItemsFlexStart-11"
    disabled={false}
  >
    <div
      className="MuiAvatar-root-21 MuiListItemAvatar-alignItemsFlexStart-19"
    >
      <img
        alt="Pedro Santiago"
        className="MuiAvatar-img-23"
        src="https://source.unsplash.com/2ShvY8Lf6l0/800x599"
      />
    </div>
    <div
      className="MuiListItemText-root-24"
    >
      <span
        className="MuiTypography-root-30 MuiTypography-subheading-37 MuiListItemText-primary-27"
      >
        Pedro Santiago
      </span>
      <p
        className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextSecondary-63 MuiListItemText-secondary-28"
      >
        <span
          className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextPrimary-62 MuiTypography-inline-65"
        >
          This is a bad message
        </span>
        <br />
        2019-3-21 12:00:00
      </p>
    </div>
  </li>
  <li
    className="MuiListItem-root-6 MuiListItem-default-9 MuiListItem-dense-10 MuiListItem-gutters-14 MuiListItem-alignItemsFlexStart-11"
    disabled={false}
  >
    <div
      className="MuiAvatar-root-21 MuiListItemAvatar-alignItemsFlexStart-19"
    >
      <img
        alt="Pedro Rivera"
        className="MuiAvatar-img-23"
        src="https://source.unsplash.com/2ShvY8Lf6l0/800x599"
      />
    </div>
    <div
      className="MuiListItemText-root-24"
    >
      <span
        className="MuiTypography-root-30 MuiTypography-subheading-37 MuiListItemText-primary-27"
      >
        Pedro Rivera
      </span>
      <p
        className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextSecondary-63 MuiListItemText-secondary-28"
      >
        <span
          className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextPrimary-62 MuiTypography-inline-65"
        >
          This is an awesome message
        </span>
        <br />
        2019-3-21 12:00:00
      </p>
    </div>
  </li>
</ul>
`;
